include/dlaf/sender/traits.h

```diff
@@ -19,6 +22,58 @@ namespace dlaf::internal {
 template <typename...>
 struct TypeList {};
 
+template <typename... Ts>
+using DecayedTypeList = TypeList<std::decay_t<Ts>...>;
+
+template <typename T>
+struct IsFalse : std::integral_constant<bool, !(bool) T::value> {};
+
+template <typename T>
+inline constexpr bool IsFalseValue = IsFalse<T>::value;
+
+template <typename... Ts>
+struct AlwaysFalse : std::false_type {};
+
+template <typename... Ts>
+static std::true_type AnyOfImpl(...);
+
+template <typename... Ts>
+static auto AnyOfImpl(int) -> AlwaysFalse<std::enable_if_t<IsFalseValue<Ts>>...>;
+
+template <typename... Ts>
+struct AnyOf : decltype(AnyOfImpl<Ts...>(0)) {};
+
+template <>
+struct AnyOf<> : std::false_type {};
+
+template <typename T, typename... Ts>
+struct Contains : AnyOf<std::is_same<T, Ts>...> {};
+
+template <typename PackUnique, typename PackRest>
+struct UniqueHelper;
+
+template <template <typename...> class Pack, typename... Ts>
+struct UniqueHelper<Pack<Ts...>, Pack<>> {
+  using type = Pack<Ts...>;
+};
+
+template <template <typename...> class Pack, typename... Ts, typename U, typename... Us>
+struct UniqueHelper<Pack<Ts...>, Pack<U, Us...>>
+    : std::conditional<Contains<U, Ts...>::value, UniqueHelper<Pack<Ts...>, Pack<Us...>>,
+                       UniqueHelper<Pack<Ts..., U>, Pack<Us...>>>::type {};
+
+template <typename Pack>
+struct Unique;
+
+template <template <typename...> class Pack, typename... Ts>
+struct Unique<Pack<Ts...>> : UniqueHelper<Pack<>, Pack<Ts...>> {};
+
+/// Remove duplicate types in the given pack.
+template <typename Pack>
+using UniqueType = typename Unique<Pack>::type;
+
+struct EmptyEnv {};
```

Comment:
Just for curiosity:
Why are these struct needed for stdexec?

Reply:
It's technically also needed when stdexec is disabled in pika _in the general case_. However, practically we don't need the decaying in the use cases that currently exist in DLA-Future.

The reason is that some stdexec sender adaptors send completion signatures with `T&&` where pika would send `T` and we need to make sure we treat the two the same (as `T`).